### PR TITLE
Providing logger to instance_generator, fixes #142

### LINKED ISSUE #142: undefined local variable or method `logger' on kitchen create
Got following error under Ruby 2.2.2, test-kitchen 1.4.0, kitchen-ec2 2-0.9.3 anyone saw this or come up with work around?
```
$ kitchen create apache
-----> Starting Kitchen (v1.4.0)
-----> Creating <test-w-apache-ubuntu-1204>...
       Creating <>...
If you are not using an account that qualifies under the AWS
free-tier, you may be charged to run these suites. The charge
should be minimal, but neither Test Kitchen nor its maintainers
are responsible for your incurred costs.

>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #create action: [undefined local variable or method `logger' for #<Kitchen::Driver::Aws::InstanceGenerator:0x007fa8f01818e8>]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

Following was found in log file:
```
Kitchen: ------Backtrace-------
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/logging.rb:44:in `block in logger_method'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/aws/instance_generator.rb:139:in `block in debug_if_root_device'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/aws/instance_generator.rb:137:in `each'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/aws/instance_generator.rb:137:in `find'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/aws/instance_generator.rb:137:in `debug_if_root_device'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/aws/instance_generator.rb:120:in `block_device_mappings'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/aws/instance_generator.rb:56:in `ec2_instance_data'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/ec2.rb:297:in `submit_server'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/kitchen-ec2-0.9.3/lib/kitchen/driver/ec2.rb:188:in `create'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:424:in `public_send'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:424:in `block in perform_action'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:488:in `call'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:488:in `synchronize_or_call'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:453:in `block in action'
Kitchen: /home/Joel/.rubies/ruby-2.2.2/lib/ruby/2.2.0/benchmark.rb:288:in `measure'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:452:in `action'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:424:in `perform_action'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:352:in `create_action'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:341:in `block in transition_to'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:340:in `each'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:340:in `transition_to'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/instance.rb:127:in `create'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/command.rb:176:in `public_send'
Kitchen: /home/Joel/.gem/ruby/2.2.2/gems/test-kitchen-1.4.0/lib/kitchen/command.rb:176:in `block (2 levels) in run_action'
Kitchen: ----------------------

```

### PR DESCRIPTION
**Pull request by tyler-ball**
fixes https://github.com/test-kitchen/kitchen-ec2/issues/142

Also brought the `availability_zone` behavior in line with the README

\cc @joelhandwell @fnichol @metadave @randomcamel

### CODE REVIEW

File: lib/kitchen/driver/ec2.rb [hunk 1]
```diff
@@ -159,6 +159,8 @@ def finalize_config!(instance)
 
         if config[:availability_zone].nil?
           config[:availability_zone] = config[:region] + "b"
+        elsif config[:availability_zone] =~ /^[a-zA-Z]$/
```
Comment (metadave) [inline lib/kitchen/driver/ec2.rb]: is this related to logging?

Comment (tyler-ball) [inline lib/kitchen/driver/ec2.rb]: Not at all, it just brings the behavior in line with what was already in the README

**Comment by metadave** [comment]
+1, looks good if Travis is happy

**Comment by joelhandwell** [comment]
@tyler-ball Thank you for quick response, you are saving my day.

**Comment by fnichol** [comment]
Awesome! :+1:

**Comment by airom** [comment]
@tyler-ball THANK YOU:) :+1:

**Comment by tyler-ball** [comment]
0.9.4 released

**Comment by joelhandwell** [comment]
Thanks!

**Pull request merged**